### DISCUSSION TITLE
How to Recover TextMeshPro Components in Scenes

### DISCUSSION THREAD

**Discussion by limeraiin** [AssetRipper/AssetRipper#1385] [Q&A]
Hello. I've been using AssetRipper for a while but there is a problem I encounter regularly: when I open a decompiled project, all the TextMeshPro components in the scenes become "missing script".

Possible solutions I've tried:
1- Removing Unity's default TextMeshPro and UnityEngine.UI packages and using the decompiled (DLL) ones.
2- The opposite of the solution 1
3- Creating a fresh unity project and importing the decompiled project files there

I hope you can help me find a way to keep the text objects in scenes usable. Thanks in advance!

**Comment by ds5678** [comment]
In regards to number one, AssetRipper output shouldn't reference the TextMeshPro package, so I don't know how you did that.

**Reply by limeraiin** [reply]
Thanks for the reply. Sorry for the wording. 

I tried using AssetRipper's output (with its own UnityEngine.TextMeshPro.dll) and Unity's own TextMeshPro package separately. However, I keep getting a "missing script" warning on all TextMeshPro components in the scenes.

What is the best way to decompile a project without losing the text objects in the scenes? Because even if I open the output folder as a project, I lose all the textmeshpro references in scenes.

The project is from 2019.4.18f1

**Reply by ds5678** [reply]
You probably have script errors. Unity won't show anything until those are resolved.

**Reply by limeraiin** [reply]
> You probably have script errors. Unity won't show anything until those are resolved.

But there are no errors and I can enter the play mode. I will try it again today. Should I create a new project and import the output of Assetripper into it or should I open the output as a project?

Edit: maybe the guids need to be remapped but not sure how to do that.

**Reply by StupidRepo** [reply]
Yes I am having this issue too and yes it has something to do with the GUIDs. Did you ever figure it out?
